cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/config/BrokerSetFileResolverTest.java

```diff
@@ -72,17 +73,28 @@ public void testNonExistentFile() {
                AnalyzerConfig.BROKER_SET_CONFIG_FILE_CONFIG, fileName,
                BrokerSetFileResolver.BROKER_SET_ASSIGNMENT_POLICY_OBJECT_CONFIG, new NoOpBrokerSetAssignmentPolicy());
     brokerSetResolver.configure(configs);
-    assertThrows(IllegalArgumentException.class,
-                 () -> brokerSetResolver.brokerIdsByBrokerSetId(DeterministicCluster.brokerSetSatisfiable1()));
+    assertThrows(BrokerSetResolutionException.class,
+                 () -> brokerSetResolver.brokerIdsByBrokerSetId(
+                     BrokerSetResolutionHelper.getRackIdByBrokerIdMapping(DeterministicCluster.brokerSetSatisfiable1())));
   }
 
   /**
-   * Tests if the cluster model has extra brokers that are unresolved from broker set mapping then throws broker set resoltuion exception
+   * Tests if the cluster model has extra brokers that are unresolved from broker set mapping then throws broker set resolution exception
    */
   @Test
-  public void testBrokerSetResolutionExceptionWithNoOpDefaultAssignmentPolicy() {
+  public void testBrokerSetResolutionWithNoOpDefaultAssignmentPolicy() throws BrokerSetResolutionException {
     BrokerSetResolver brokerSetResolver = getBrokerSetResolver("testBrokerSets.json", this.getClass());
-    assertThrows(BrokerSetResolutionException.class,
-                 () -> brokerSetResolver.brokerIdsByBrokerSetId(DeterministicCluster.brokerSetUnSatisfiable3()));
+    final Map<String, Set<Integer>> brokerSets = brokerSetResolver.brokerIdsByBrokerSetId(
+        BrokerSetResolutionHelper.getRackIdByBrokerIdMapping(DeterministicCluster.brokerSetUnSatisfiable3()));
```

Comment:
I dont remember why it was throwing Exception earlier.

Reply:
It throws exception if there is any broker that is not assigned with broker set. Now the NoOp policy will make sure that all brokers will be assigned with at least "unmapped" broker id. So now there shouldn't be any exceptions thrown with NoOp policy.